design-documents/frontend-html-binding.md

```diff
@@ -0,0 +1,18 @@
+### Rule for binding HTML values using knockout/Magento syntax
+#### Description
+Templates allow binding javascript variables/expressions using knockout/Magento to DOM elements' HTML
+like this:
+```html
+<!-- Knockout syntax -->
+<div data-bind="html: variable"></div>
+<!-- Magento syntax -->
+<div html="variable"></div>
+```
+Developers may use this binding instead of text binding for all values leaving XSS
+vulnerabilities.
+
+#### Proposing
+Create static test to require developers to use variables/functions with _'html'_ suffix
+to bind HTML (like _variableHtml_ or _widget.generateButtonHtml()_). It's not a 100% guaranty that properly sanitized values will be used for dynamic HTML
+but it will notify developers to think whether they really need HTML binding for a value or
```

Comment:
In most cases, 3rd party developers don't run static tests, so for them, it won't work.

Reply:
extension developers and community will run the tests - I think it's better than nothing